docs/section21/Rule21-7.md

```diff
@@ -29,7 +30,7 @@
 
 ## Rule Logic:  
 
-- For each boiler in B_RMR, save boiler to loop boiler dictionary: `for boiler_b in B_RMR.RulesetModelInstance.boilers: loop_boiler_dict[boiler_b.loop].append(boiler_b)`
+- For each boiler in B_RMI, save boiler to loop boiler dictionary: `loop_boiler_dict = find_all(B-RMI)`
```

Comment:
The modification does not seems to save boilers to loop boiler dictionary.

If this is a list, we should explicitly called this out as a list.

Reply:
One more comment is: `find_all` is a utility function. Generally, in the RDS we only put `ruleset specific utility` function. A utility function is usually expanded in the RDS or give a more explicit explanations.
In this case, the line can be updated to:

```
loop_boiler_ids = find_all(boiler.id, B_RMI)
```
or
```
for boiler_b in B_RMI.RulesetModelInstance.boilers:
    loop_boiler_ids.append(boiler_b.id)
```

Reply:
Fixed. Thanks for letting me know about this Weili.